nbgrader/apps/baseapp.py

```diff
@@ -160,6 +165,110 @@ def __init__(self, *args, **kwargs):
         super(BaseNbGraderApp, self).__init__(*args, **kwargs)
         self._nbgrader_config = NbGraderConfig(parent=self)
 
+        
+# These are the aliases and flags for nbgrader apps that inherit only from
+# TransferApp
+transfer_aliases = {}
+transfer_aliases.update(nbgrader_aliases)
+transfer_aliases.update({
+    "timezone": "TransferApp.timezone"
+})
+transfer_flags = {}
+transfer_flags.update(nbgrader_flags)
+transfer_flags.update({
+})
+        
+class TransferApp(BaseNbGraderApp):
+    """A base class for the release, collect, fetch and submit apps.
+    
+    All of these apps involve transfering files between an instructor or students
+    files and the nbgrader exchange.
+    """
+    
+    timezone = Unicode(
+        "UTC", config=True,
+        help="Timezone for recording timestamps"
+    )
+
+    timestamp_format = Unicode(
+        "%Y-%m-%d %H:%M:%S %Z", config=True,
+        help="Format string for timestamps"
+    )
+
+    def set_timestamp(self):
+        """Set the timestap."""
+        tz = gettz(self.timezone)
+        if tz is None:
+            self.fail("Invalid timezone: {}".format(self.timezone))
+        self.timestamp = datetime.datetime.now(tz).strftime(self.timestamp_format)
+    
+    ignore = List(
```

Comment:
It might be worth moving this up a level into `BaseNbGraderApp`, although not all apps that inherit from `BaseNbGraderApp` will use it. But I also put an `ignore` configurable in `BaseNbConvertApp` in #203 , so perhaps rather than having it in both the transfer app and the nbconvert app, we just have it in one place.